CDP4IME/CDP4IMEBootstrapper.cs

```diff
@@ -61,31 +65,66 @@ protected override DependencyObject CreateShell()
         /// </summary>
         protected override void InitializeShell()
         {
+            var appSettings = this.Container.GetExportedValue<IAppSettingsService>();
+            var appServiceSettings = this.ReadAppSettings(appSettings);
+
             this.UpdateBootstrapperState("Loading CDP4 Plugins");
             var pluginCatalog = new CDP4PluginLoader();
             foreach (var directoryCatalog in pluginCatalog.DirectoryCatalogues)
             {
-                this.AggregateCatalog.Catalogs.Add(directoryCatalog);
+                var fileName = Path.GetFileName(directoryCatalog.FullPath);
```

Comment:
i think putting the logic here is to late. The `CDP4PluginLoader` does plugin loading, so you need to provide that class with a list of Pluigins to load or not load. The nice thing is that this code does not need to be repeated for the IME and the Add-In.

Reply:
Agreed, the settings have to be injected there.

Reply:
Logic gas been moved to PluginLoader